lightningd/channel.c

```diff
@@ -619,11 +619,36 @@ static void htlc_incstate(struct channel *channel,
 	}
 }
 
+static void check_lockedin(const struct htlc *h,
+			   struct peer *peer,
+			   void (*oursfail)(struct peer *peer,
```

Comment:
Yeah.  In theory, they're really generic callbacks at this layer.

In practice, instead of passing through void * (or using ccan/typesafe_cb) we pass through a 'struct peer' because I'm lazy.  I'll fix to make it cleaner.